### PR TITLE
1st set of SRX rules

### PR DESCRIPTION
**Pull request by sraju4ug**
System, Linecard and security rules of SRX

### CODE REVIEW

**Comment by gvarao** [comment]
jenkins please retry a build

File: juniper_official/SRX/Security/srx-cluster-flow-session-summary.rule [hunk 1]
```diff
@@ -0,0 +1,225 @@
+/*
+ * Detects flow session details for each PIC and notifies when anomalies are found.
+ * 
+ * 
+ *   1) "active-sessions-threshold-value" is the threshold that causes the rule to report
+ *      an anomaly.  By default it's 5000000. This rule will set a dashboard
+ *      color to red when active sessions for a PIC slot is greater than this value.
+ *   2) "max-sessions-threshold-value" is the threshold that causes the rule to report
+ *      an anomaly.  By default it's 5000000. This rule will set a dashboard
```
Comment (gvarao) [inline juniper_official/SRX/Security/srx-cluster-flow-session-summary.rule]: Plesse check "max-sessions-threshold-value", it should be 8000000

File: juniper_official/SRX/Security/srx-cluster-flow-session-summary.rule [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+ *      color to red when active sessions for a PIC slot is greater than this value.
+ *  
+ */
+iceberg {
+    topic security  {
+        rule check-flow-session-summery {
+            description "Monitors and notify flow session details";
+            synopsis "flow session summery statistics analyzer";
```
Comment (gvarao) [inline juniper_official/SRX/Security/srx-cluster-flow-session-summary.rule]: - Start descriptions and synopsis with CAPS letter in all rules
- s/summery/summary/g

File: juniper_official/SRX/Security/srx-cluster-flow-session-summary.rule [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+ *      color to red when active sessions for a PIC slot is greater than this value.
+ *  
+ */
+iceberg {
+    topic security  {
+        rule check-flow-session-summery {
+            description "Monitors and notify flow session details";
+            synopsis "flow session summery statistics analyzer"; 
+            /*
+             * Monitors the active flow session setails  and notifies via the dashboard
```
Comment (gvarao) [inline juniper_official/SRX/Security/srx-cluster-flow-session-summary.rule]: s/setails  /details /g

File: juniper_official/SRX/Security/srx-cluster-flow-session-summary.rule [hunk 4]
```diff
@@ -0,0 +1,225 @@
...
+             * colors when the threshold is above {{active-sessions-threshold}}
+             * it will mark as red.
+             * The color is set to yellow when the traffic is above dt-active-sessions.
```
Comment (gvarao) [inline juniper_official/SRX/Security/srx-cluster-flow-session-summary.rule]: s/dt-active-sessions/dynamic threshold {{dt-active-sessions}}/g

File: juniper_official/SRX/Security/srx-cluster-flow-session-summary.rule [hunk 5]
```diff
@@ -0,0 +1,225 @@
...
+             * Otherwise the color is set to green.
+             * 
+             * Monitors the maximimum flow session setails  and notifies via the dashboard
```
Comment (gvarao) [inline juniper_official/SRX/Security/srx-cluster-flow-session-summary.rule]: s/setails /details /g

File: juniper_official/SRX/Security/srx-cluster-flow-session-summary.rule [hunk 6]
```diff
@@ -0,0 +1,225 @@
...
+             * colors when the threshold is above {{max-sessions-threshold}}
+             * it will mark as red.
+             * The color is set to yellow when the traffic is above dt-max-sessions.
```
Comment (gvarao) [inline juniper_official/SRX/Security/srx-cluster-flow-session-summary.rule]: s/dt-active-sessions/dynamic threshold {{dt-max-sessions}}/g

File: juniper_official/SRX/Security/srx-cluster-flow-session-summary.rule [hunk 7]
```diff
@@ -0,0 +1,225 @@
...
+                formula {
+                    dynamic-threshold {
+                        algorithm 3sigma;
+                        learning-period 7d;
+                        pattern-periodicity 1d,1h;
+                        field-name "$active-sessions";
+                    }
+                }
+                type integer;
+                description "This filed contains the active flow sessions threshold value";
```
Comment (gvarao) [inline juniper_official/SRX/Security/srx-cluster-flow-session-summary.rule]: s/"This filed contains the active flow sessions threshold value"/"ML derived active flow sessions threshold value"/g

File: juniper_official/SRX/Security/srx-cluster-flow-session-summary.rule [hunk 8]
```diff
@@ -0,0 +1,225 @@
...
+                formula {
+                    dynamic-threshold {
+                        algorithm 3sigma;
+                        learning-period 7d;
+                        pattern-periodicity 1d,1h;
+                        field-name "$max-sessions";
+                    }
+                }
+                type integer;
+                description "This filed contains the maximum flow sessions threshold value";
```
Comment (gvarao) [inline juniper_official/SRX/Security/srx-cluster-flow-session-summary.rule]: s/"This filed contains the maximum flow sessions threshold value";/"ML derived maximum flow sessions threshold value";/g

File: juniper_official/SRX/Security/srx-cluster-flow-session-summary.rule [hunk 9]
```diff
@@ -0,0 +1,225 @@
...
+                    path re-name;
+                }
+                type string;
+                description "This field contains node details";
+            }
+            /*
+             * Anomaly detection logic for active flow sessions
+             */
+            trigger active-sessions {
+	        frequency 60s;
```
Comment (gvarao) [inline juniper_official/SRX/Security/srx-cluster-flow-session-summary.rule]: It is good to have trigger frequecy as minimum 1.5 times of sensor frequency.

File: juniper_official/SRX/Security/srx-cluster-flow-session-summary.rule [hunk 10]
```diff
@@ -0,0 +1,225 @@
...
+                term is-active-sessions-abnormal {
+                    when {
+                        greater-than-or-equal-to "$active-sessions" "$active-sessions-threshold" {
+                            time-range 180s;
+                        }
+                    }
+                    then {
+                        status {
+                            color red;
+                            message "Active flow sessions on Node=$re-name & flow id = $fpc-pic-id has crossed static threshold value ($active-sessions-threshold), Value=$active-sessions";
```
Comment (gvarao) [inline juniper_official/SRX/Security/srx-cluster-flow-session-summary.rule]: s/"Active flow sessions on Node=$re-name & flow id = $fpc-pic-id has crossed static threshold value ($active-sessions-threshold), Value=$active-sessions";/"Active flow sessions($active-sessions) on Node:$re-name, flow id:$fpc-pic-id has crossed static threshold value ($active-sessions-threshold)";/g

File: juniper_official/SRX/Security/srx-cluster-flow-session-summary.rule [hunk 11]
```diff
@@ -0,0 +1,231 @@
...
+                    value "{{active-sessions-threshold-value}}";
+                }
+                type integer;
+                description "This filed contains the active flow session threshold value";
+            }
+            field dt-active-sessions {
+                formula {
+                    dynamic-threshold {
+                        algorithm 3sigma;
+                        learning-period 7d;
```
Comment (ConstrainedRandomVar) [inline juniper_official/SRX/Security/srx-cluster-flow-session-summary.rule]: Please use the new expanded notation for all learning-periods and pattern periodicities. 
For example: learning-period 7days 
pattern-periodicity 1days,1hours

**Pull request merged**